opsgenie.go

```diff
@@ -43,42 +47,57 @@ func NewOpsgenieClient(apikey string) (*OpsgenieClient, error) {
 	}, nil
 }
 
-func (c *OpsgenieClient) CountAlerts() (float64, error) {
-	res, err := c.alerts.CountAlerts(context.Background(), &alert.CountAlertsRequest{
-		Query: "",
+// countAlerts accepts a search query to limit the results returned.
+// See https://support.atlassian.com/opsgenie/docs/search-queries-for-alerts/
+func (c *OpsgenieClient) countAlerts(ctx context.Context, query string) (float64, error) {
+	res, err := c.alerts.CountAlerts(ctx, &alert.CountAlertsRequest{
+		Query: query,
 	})
 	if err != nil {
 		return 0, err
 	}
+
 	return float64(res.Count), nil
 }
 
-func (c *OpsgenieClient) CountClosedAlerts() (float64, error) {
-	res, err := c.alerts.CountAlerts(context.Background(), &alert.CountAlertsRequest{
-		Query: fmt.Sprintf("status: %s", alertClosedStatus),
-	})
-	if err != nil {
-		return 0, err
-	}
-	return float64(res.Count), nil
+type countAlertsParams struct {
+	Team     string
+	Status   string
+	Priority string
 }
 
-func (c *OpsgenieClient) CountOpenAlerts() (float64, error) {
-	res, err := c.alerts.CountAlerts(context.Background(), &alert.CountAlertsRequest{
-		Query: fmt.Sprintf("status: %s", alertOpenStatus),
-	})
-	if err != nil {
-		return 0, err
+func (p *countAlertsParams) ToQuery() string {
+	var filters []string
+	if p.Team != "" {
+		filters = append(filters, fmt.Sprintf("teams: %s", p.Team))
 	}
-	return float64(res.Count), nil
+	if p.Status != "" {
+		filters = append(filters, fmt.Sprintf("status: %s", p.Team))
```

Comment:
Also tests were failing, for example here the wrong value for status is used (must be `p.Status` instead of `p.Team` and the outputs need adjustments because when the queries are created, the team filter always comes first but you're expecting another string in the test cases. Just fyi